### PR TITLE
[#7] [UI] Login form

### PR DESCRIPTION
**Pull request by AVI5HEK**
https://github.com/nimblehq/ic-flutter-avishek/issues/7

## What happened 👀

-  Show the splash screen when the app starts.
- When the Logo animation ends, show the Login form.

## Insight 📝

N/A.

### Android

https://github.com/nimblehq/ic-flutter-avishek/assets/8093908/e5e97452-44f0-438c-9cf2-f7f9b30588ec

### iOS
https://github.com/nimblehq/ic-flutter-avishek/assets/8093908/fc24c00b-989c-4d70-b451-c3eb01b580d9

### CODE REVIEW

File: lib/ui/splash/splash_screen.dart [hunk 1]
```diff
@@ -93,3 +120,109 @@ class _SplashScreenState extends ConsumerState<SplashScreen>
     super.dispose();
   }
 }
+
+class LoginForm extends ConsumerStatefulWidget {
```
Comment (markgravity) [inline lib/ui/splash/splash_screen.dart:124]: As we do this approach, I think we should rename this screen to `LoginScreen`, what do you think?

Comment (AVI5HEK) [reply lib/ui/splash/splash_screen.dart]: @markgravity done in https://github.com/nimblehq/ic-flutter-avishek/pull/33/commits/12a37f81ec057868068a63706c8b15eda134e744.

File: lib/ui/splash/login_screen.dart [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+import '../widget/primary_text_form_field_decoration.dart';
+
+const _logoRevealDuration = Duration(milliseconds: 500);
+const _logoDuration = Duration(milliseconds: 750);
+const _loginFormRevealDuration = Duration(milliseconds: 700);
+
+final _shouldAnimateLogoPositionProvider =
+    StateProvider.autoDispose<bool>((_) => false);
+
+class LoginScreen extends ConsumerStatefulWidget {
```
Comment (markgravity) [inline lib/ui/splash/login_screen.dart:18]: Please update the sub folder as well, it's `splash` currently 🙏

File: lib/ui/splash/login_screen.dart [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  }
+
+  @override
+  void dispose() {
+    _logoRevealAnimationController.dispose();
+    super.dispose();
+  }
+}
+
+class LoginForm extends ConsumerStatefulWidget {
```
Comment (markgravity) [inline lib/ui/splash/login_screen.dart:124]: What do you think about moving this widget to a separate file? 🙏 
For example: `lib/ui/login/widgets/login_form.dart`

File: lib/ui/splash/login_screen.dart (hunk 2 above)
Comment (AVI5HEK) [reply lib/ui/splash/login_screen.dart]: Updated in https://github.com/nimblehq/ic-flutter-avishek/pull/33/commits/013d5f49facd12f481667505c58cc29314cd7e3d

File: lib/ui/splash/login_screen.dart (hunk 3 above)
Comment (AVI5HEK) [reply lib/ui/splash/login_screen.dart]: Updated in https://github.com/nimblehq/ic-flutter-avishek/pull/33/commits/013d5f49facd12f481667505c58cc29314cd7e3d

File: lib/ui/splash/login_screen.dart [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    );
+  }
+
+  Widget _buildLoginForm() {
+    const forgotButtonWidth = 80.0;
+
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        TextFormField(
```
[the remark this answers is not in the document]
Comment (AVI5HEK) [reply lib/ui/splash/login_screen.dart]: Updated in https://github.com/nimblehq/ic-flutter-avishek/pull/33/commits/013d5f49facd12f481667505c58cc29314cd7e3d

**Review by markgravity — APPROVED** [review]

**Review by Wadeewee — APPROVED** [review]

**Pull request merged**